vending-machine/src/components/vendingMachine/showWindow_itemBox/itemBox.js

```diff
@@ -0,0 +1,71 @@
+import { StyledItemContainer, StyledItemName, StyledItemPrice } from './itemBox.styled';
+import { getWonTemplate, delay } from '../../../helper/utils';
+import { useContext, useEffect, useState } from 'react';
+import { InputMoneyContext, LogContext } from '../vendingMachine';
+
+export function ItemBox({ item, inProgress, setInProgress }) {
+  const { inputMoney, setInputMoney } = useContext(InputMoneyContext);
+  const { logList, setLogList } = useContext(LogContext);
+  const [boxColor, setBoxColor] = useState('gray');
+  const [itemStock, setItemStock] = useState(item.stock);
+  const ITEM_DROP_TIME = 2000;
+
+  function handleClick() {
+    if (!itemStock) return;
+    if (item.price > inputMoney) return;
+    if (inProgress) return;
```

Comment:
```suggestion
    const invalidItem = !itemStock || item.price > inputMoney || inProgress;
    if (invalidItem) return;
```

같은 동작을 한다면 조건을 합쳐볼 수 있겠네요!